kwasm-lemmas.md

```diff
@@ -21,6 +21,11 @@ Basic logic
 Basic arithmetic
 ----------------
 
+```k
+    rule #signed(_, 0) => 0 [simplification]
+    rule #signed(_, X -Int Y) => X -Int Y [simplification]
```

Comment:
This isn't true. Consider the definition of `#signed` (in `data.md`)

```
    rule #signed(ITYPE, N) => N                  requires 0            <=Int N andBool N <Int #pow1(ITYPE) 
    rule #signed(ITYPE, N) => N -Int #pow(ITYPE) requires #pow1(ITYPE) <=Int N andBool N <Int #pow (ITYPE) 
```

So if we have that `notBool (0 <=Int X -Int Y andBool X - Int Y <Int #pow1(ITYPE))`, and we also have that `#pow1(ITYPE) <=Int X -Int Y andBool X -Int Y <Int #pow(ITYPE)`, we should be returning `(X -Int Y) -Int #pow(ITYPE)`, but your simplification rule returns `X -Int Y`.

Reply:
You can add a requires clause to your second rule which says `0 <=Int X -Int Y andBool X -Int Y <Int #pow1(ITYPE)`, but that would be saying the same thing as the original rule in the semantics.

So instead, we should make it so the prover can know that `0 <=Int X -Int Y andBool X -Int Y <Int #pow1(ITYPE)` is true. Which proof is this lemma needed for?